markdown_lab/network/async_client.py

```
@@ -0,0 +1,576 @@
+            url: URL to fetch
+            **kwargs: Additional request arguments
+
+        Returns:
+            Tuple of (url, content_or_exception)
+        """
+        if self._semaphore is None:
+            raise RuntimeError("Semaphore not initialized. Use within async context manager.")
+        async with self._semaphore:
+            try:
+                # Add per-domain rate limiting if configured
+                urlparse(url).netloc
```

Ruff (0.11.9)

Found useless attribute access. Either assign it to a variable or remove it.

(B018)